src/sinks/util/encoding/adapter.rs

```diff
@@ -0,0 +1,335 @@
+                        OnlyOrExceptFieldsConfig::OnlyFields(fields) => {
+                            Some(fields.only_fields.clone())
+                        }
+                        _ => None,
+                    });
+                let except_fields =
+                    config
+                        .encoding
+                        .filter
+                        .as_ref()
+                        .and_then(|filter| match filter {
+                            OnlyOrExceptFieldsConfig::ExceptFields(fields) => {
+                                Some(fields.except_fields.clone())
+                            }
+                            _ => None,
+                        });
+                let timestamp_format = config.encoding.timestamp_format;
+
+                Transformer {
+                    only_fields,
+                    except_fields,
+                    timestamp_format,
+                }
+            }
+            Self::LegacyEncodingConfig(config) => Transformer {
+                only_fields: config.encoding.only_fields().as_ref().map(|fields| {
+                    fields
+                        .iter()
+                        .map(|field| {
+                            field
+                                .iter()
+                                .map(|component| component.clone().into_static())
+                                .collect()
+                        })
+                        .collect()
+                }),
+                except_fields: config.encoding.except_fields().clone(),
+                timestamp_format: *config.encoding.timestamp_format(),
+            },
+        }
+    }
+
+    /// Build the framer and serializer for this configuration.
+    pub fn encoding(&self) -> crate::Result<(Option<Box<dyn Framer>>, Box<dyn Serializer>)> {
+        let (framer, serializer) = match self {
+            Self::Encoding(config) => {
+                let framer = match &config.framing {
+                    Some(framing) => Some(framing.build()?),
+                    None => None,
+                };
+                let serializer = config.encoding.encoding.build()?;
+
+                (framer, serializer)
+            }
+            Self::LegacyEncodingConfig(config) => {
+                let migration = Migrator::migrate(config.encoding.codec());
+                let framer = match migration.0 {
+                    Some(framing) => Some(framing.build()?),
+                    None => None,
+                };
+                let serializer = migration.1.build()?;
+
+                (framer, serializer)
+            }
+        };
+
+        Ok((framer, serializer))
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LegacyEncodingConfigWrapper<EncodingConfig, Migrator> {
+    encoding: EncodingConfig,
+    #[serde(skip)]
+    phantom: PhantomData<Migrator>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct EncodingConfig {
```

Comment:
I don't love that we have `EncodingConfiguration` and `EncodingConfig`. Seems like one of them needs a better name.

Reply:
@lukesteensen fully agree - however that's just piling on the naming convention we currently have on master:

https://github.com/vectordotdev/vector/blob/9c136c7cddd48f0768c1e0134411a55607457333/src/sinks/util/encoding/config.rs#L38

Reply:
Will that be able to go away once we've migrated more fully to the new system? Might be worth spinning off an issue or something.

Reply:
Yep, makes sense to tackle once we fully migrated. Opened https://github.com/vectordotdev/vector/issues/10811.